django/db/migrations/state.py

```diff
@@ -46,27 +56,100 @@ def remove_model(self, app_label, model_name):
 
     def reload_model(self, app_label, model_name):
         if 'apps' in self.__dict__:  # hasattr would cache the property
-            # Get relations before reloading the models, as _meta.apps may change
             try:
-                related_old = {
-                    f.related_model for f in
-                    self.apps.get_model(app_label, model_name)._meta.related_objects
-                }
+                old_model = self.apps.get_model(app_label, model_name)
             except LookupError:
-                related_old = set()
-            self._reload_one_model(app_label, model_name)
-            # Reload models if there are relations
-            model = self.apps.get_model(app_label, model_name)
-            related_m2m = {f.related_model for f in model._meta.many_to_many}
-            for rel_model in related_old.union(related_m2m):
-                self._reload_one_model(rel_model._meta.app_label, rel_model._meta.model_name)
-            if related_m2m:
-                # Re-render this model after related models have been reloaded
-                self._reload_one_model(app_label, model_name)
+                related_models = set()
+            else:
+                # Get all relations to and from the old model before reloading,
+                # as _meta.apps may change
+                related_models = ProjectState._related_models_recursive(old_model)
+
+            # Get all outgoing references from the model to be rendered
+            model_state = self.models[(app_label, model_name)]
+            for name, field in model_state.fields:
+                if field.is_relation:
+                    if field.rel.to == RECURSIVE_RELATIONSHIP_CONSTANT:
+                        continue
+                    al, mn = _get_app_label_and_model_name(field.rel.to, app_label)
```

Comment:
I'd like longer variable names so there's no cognitive load of expanding abbrevations